exporters/zipkin/src/main/java/io/opentelemetry/exporter/zipkin/ZipkinSpanExporterBuilder.java

```diff
@@ -38,20 +38,35 @@ public final class ZipkinSpanExporterBuilder {
    * Sets the Zipkin sender. Implements the client side of the span transport. An {@link
    * OkHttpSender} is a good default.
    *
-   * <p>The {@link Sender#close()} method will be called when the exporter is shut down.
+   * <p>The {@link BytesMessageSender#close()} method will be called when the exporter is shut down.
    *
    * @param sender the Zipkin sender implementation.
    * @return this.
+   * @deprecated Use {@link #setSender(BytesMessageSender)} insteead.
    */
-  public ZipkinSpanExporterBuilder setSender(Sender sender) {
+  @Deprecated
+  public ZipkinSpanExporterBuilder setSender(zipkin2.reporter.Sender sender) {
```

Comment:
The javadoc says that this will be removed in version 4. @open-telemetry/java-approvers we should figure out a plan with how we would like to handle this. If we bump to the next major version of zipkin, we wouldn't be able to support this API.

Reply:
FWIW I'm also not sure when we would do a version 4, but the point remains. The upgrade analysis forced this point, but I think critical thinking could make it easier.

Reporter 3.x isn't compatible with 2.x senders anyway. You haven't yet released the 3.x code yet.... one could argue you could remove this deprecated thing now and then not have to worry about v4.

Reply:
for context micrometer and spring boot are waiting for 1.35.0, and that's the first version that has any of the reporter 3 stuff. They will switch all the senders to it (I will help just to help). I know you have other concerns besides spring, but it is a data point that may help you decide.

Reply:
> You haven't yet released the 3.x code yet.... one could argue you could remove this deprecated thing now and then not have to worry about v4.

Its a good point. We would have to do the same strategy I'm proposing - publish a new artifact id which includes zipkin 3 in the name some how. Deprecate, and eventually stop publishing the existing zipkin exporter artifact. 

Added a topic to discuss in the tomorrow's java SIG.

Reply:
yep another choice for discussion is to shade it and just expose it as configuration instead of a library hook. anyway thanks for the help!

Reply:
@jack-berg do you have an idea which way you are going to go? If this doesn't merge it isn't dire, just I will need to help split micrometer's classpath as the only place this causes a conflict in is tests. I mean it will be a conflict in real life, too, but only for those simultaneously using otel and also direct zipkin stuff.

Reply:
I'm inclined to merge this PR and continue to publish the artifact under the coordinates. Then, we can consider publishing new artifact coordinates when zipkin publishes a v4 with the breaking change. Solving the problem now seems like putting the cart ahead of the horse. 

@open-telemetry/java-approvers if there are no dissenting additional thoughts on this I'll plan on merging on 1/24/24.

Reply:
@jack-berg sounds good to me, and helps others from having more supporting work than maybe they can budget. v4 was just in the javadoc to tell people to stop using the old stuff as I noticed the senders were used outside of the reporters in that project. I made an issue about v4 and to not do that until it is important, likely with a new groupId/package https://github.com/openzipkin/zipkin-reporter-java/issues/247

Reply:
> I'm inclined to merge this PR and continue to publish the artifact under the coordinates. Then, we can consider publishing new artifact coordinates when zipkin publishes a v4 with the breaking change. Solving the problem now seems like putting the cart ahead of the horse.

👍